### PR TITLE
[Merged by Bors] - Added create_log4j_2_config to product logging

### PR DESCRIPTION
**Pull request by maltesander**
## Description

- Added method to create log4j2 config properties

## Review Checklist
- [x] Code contains useful comments
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,9 +10,11 @@ All notable changes to this project will be documented in this file.
 
 - Extended the `LdapAuthenticationProvider` with functionality to build add Volumes and Mounts to PodBuilder and ContainerBuilder ([#535]).
 - Extended the `PodBuilder` with `add_volume_with_empty_dir` utility function ([#536]).
+- Added method to create log4j2 config properties to product logging ([#540]).
```
Comment (siegfriedweber) [inline CHANGELOG.md:13]: This entry belongs to the [Unreleased] section.

File: src/product_logging/framework.rs [hunk 2]
```diff
@@ -244,6 +244,151 @@ log4j.appender.FILE.layout=org.apache.log4j.xml.XMLLayout
...
+            format!(
+                "logger.{name}.name = {name}\nlogger.{name}.level = {level}\n",
+                name = name.escape_default(),
+                level = logger_config.level.to_log4j_literal(),
+            )
+        })
+        .collect::<String>();
+
+    format!(
+        r#"
```
Comment (siegfriedweber) [inline src/product_logging/framework.rs:346]: **nit:** unnecessary newline

The resulting YAML looks like this:
```
  log4j.properties: |2-

    appenders = FILE, CONSOLE
```

File: src/product_logging/framework.rs [hunk 3]
```diff
@@ -244,6 +244,151 @@ log4j.appender.FILE.layout=org.apache.log4j.xml.XMLLayout
...
+appender.FILE.policies.size.size = {max_log_file_size_in_mib}MB
+appender.FILE.strategy.type = DefaultRolloverStrategy
+appender.FILE.strategy.max = {number_of_archived_log_files}
+appender.FILE.filter.threshold.type = ThresholdFilter
+appender.FILE.filter.threshold.level = {file_log_level}
+{loggers}
+{logger_configs}
+rootLogger.level={root_log_level}
+rootLogger.appenderRefs = root
+rootLogger.appenderRef.root.ref = FILE"#,
```
Comment (siegfriedweber) [inline src/product_logging/framework.rs:373]: The reference to the console appender is missing, isn't it?

```suggestion
rootLogger.appenderRefs = CONSOLE, FILE
rootLogger.appenderRef.CONSOLE.ref = CONSOLE
rootLogger.appenderRef.FILE.ref = FILE"#,
```

**Review by siegfriedweber — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (maltesander) [reply CHANGELOG.md]: Oh good find! https://github.com/stackabletech/operator-rs/pull/540/commits/001176e0abfd21514f29c16116ab3c7949c06c91

File: src/product_logging/framework.rs (hunk 2 above)
Comment (maltesander) [reply src/product_logging/framework.rs]: https://github.com/stackabletech/operator-rs/pull/540/commits/d88c30565464a017310770c7c623f02c24a3ad74

**Review by siegfriedweber — APPROVED** [review]
LGTM

**Comment by maltesander** [comment]
bors merge

**Pull request closed without merging**